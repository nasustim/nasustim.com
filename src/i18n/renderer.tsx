import { Link } from '@/components/texts/Link'
import { FC } from 'react'
import { useCallback } from 'react'
import { ReactNode } from 'react'
import { I18nInlineContent, I18nInlineValue } from './type'

export const Inline: FC<{ content: I18nInlineContent }> = ({ content }) => {
  const inlineContent = useCallback((value: I18nInlineValue[number], index: number) => {
    if (typeof value === 'string') {
      const result = Array<ReactNode>()
      const lines = value.split('\n')
      for (let j = 0; j < lines.length; j++) {
        result.push(lines[j])
        if (j + 1 !== lines.length) result.push(<br key={`inline-linebreak-${j}`} />)
      }

      return result
    }

    return <Inline content={value} key={`inline-content-${index}`} />
  }, [])

  switch (content.type) {
    case 'p':
      return content.value.map(inlineContent)
    case 'a':
      return (
        <Link href={content.href.toString()} ariaLabel={content.areaLabel}>
          {content.value.map(inlineContent)}
        </Link>
      )
  }
}
