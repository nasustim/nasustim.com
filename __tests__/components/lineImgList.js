import React from 'react'
import renderer from 'react-test-renderer'

import LineImgList from 'nasustim.com/src/components/lineImgList'

describe('LineImgList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<LineImgList device={'desktop'} works={[{ linkUri: '', title: '', imgSrc: '' }]} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
