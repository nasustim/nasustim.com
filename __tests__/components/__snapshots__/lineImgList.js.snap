// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineImgList renders correctly 1`] = `
<div
  className="sc-hKgILt gxdpBG"
>
  <section
    className="sc-eCssSg hTpebd"
  >
    <a
      className="sc-jSgupP btasya"
      href=""
    >
      <figure
        className="sc-dlfnbm gHjUtb"
      >
        <img
          alt=""
          className="sc-bdfBwQ gQxvEX"
          src=""
        />
        <figcaption
          className="sc-gsTCUz UlJRq"
        >
          <div>
            <h1>
              
            </h1>
          </div>
        </figcaption>
      </figure>
    </a>
  </section>
</div>
`;
